diff --git a/src/accounts/mod.rs b/src/accounts/mod.rs
index ac1445e..d48c364 100644
--- a/src/accounts/mod.rs
+++ b/src/accounts/mod.rs
@@ -6,6 +6,7 @@ pub mod registered_stake;
 pub mod staking_account;
 pub mod staking_vars;
 
+#[derive(Debug, serde::Serialize)]
 pub enum AtlasStakingAccount {
     RegisteredStake(registered_stake::RegisteredStake),
     StakingAccount(staking_account::StakingAccount),
