diff --git a/src/instructions/mod.rs b/src/instructions/mod.rs
index f6010ae..4877b1a 100644
--- a/src/instructions/mod.rs
+++ b/src/instructions/mod.rs
@@ -68,7 +68,7 @@ pub use self::update_recipe::*;
 pub use self::update_recipe_category::*;
 
 #[derive(Debug, Clone, PartialEq)]
-#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "serde", derive(carbon_core::InstructionType, serde::Serialize, serde::Deserialize))]
 #[cfg_attr(feature = "serde", serde(tag = "type", content = "data"))]
 pub enum CraftingInstruction {
     AddConsumableInputToRecipe(AddConsumableInputToRecipe),
