diff --git a/src/instructions/mod.rs b/src/instructions/mod.rs
index f2dd3ef..f06b095 100644
--- a/src/instructions/mod.rs
+++ b/src/instructions/mod.rs
@@ -20,7 +20,7 @@ pub use self::register_sft_redemption::*;
 pub use self::update_pack_tiers::*;
 
 #[derive(Debug, Clone, PartialEq)]
-#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "serde", derive(carbon_core::InstructionType, serde::Serialize, serde::Deserialize))]
 #[cfg_attr(feature = "serde", serde(tag = "type", content = "data"))]
 pub enum CrewInstruction {
     MintCrewMember(MintCrewMember),
