diff --git a/src/accounts/mod.rs b/src/accounts/mod.rs
index 7c98ce9..7ebdb96 100644
--- a/src/accounts/mod.rs
+++ b/src/accounts/mod.rs
@@ -6,6 +6,7 @@ pub mod points_store;
 pub mod redemption_config;
 pub mod user_redemption;
 
+#[derive(Debug, serde::Serialize)]
 pub enum PointsStoreAccount {
     PointsStore(points_store::PointsStore),
     RedemptionConfig(redemption_config::RedemptionConfig),
