diff --git a/src/accounts/mod.rs b/src/accounts/mod.rs
index b64fb2b..42f2d73 100644
--- a/src/accounts/mod.rs
+++ b/src/accounts/mod.rs
@@ -6,6 +6,7 @@ pub mod point_category;
 pub mod points_modifier;
 pub mod user_points_account;
 
+#[derive(Debug, serde::Serialize)]
 pub enum PointsAccount {
     PointCategory(point_category::PointCategory),
     PointsModifier(points_modifier::PointsModifier),
