diff --git a/src/instructions/mod.rs b/src/instructions/mod.rs
index 6c10595..a5744ee 100644
--- a/src/instructions/mod.rs
+++ b/src/instructions/mod.rs
@@ -220,7 +220,7 @@ pub use self::withdraw_cargo_from_game::*;
 pub use self::withdraw_crafting_ingredient::*;
 
 #[derive(Debug, Clone, PartialEq)]
-#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "serde", derive(carbon_core::InstructionType, serde::Serialize, serde::Deserialize))]
 #[cfg_attr(feature = "serde", serde(tag = "type", content = "data"))]
 pub enum SageInstruction {
     ActivateGameState(ActivateGameState),
