diff --git a/src/instructions/close_upgrade_process.rs b/src/instructions/close_upgrade_process.rs
index 5a0b712..dce7258 100644
--- a/src/instructions/close_upgrade_process.rs
+++ b/src/instructions/close_upgrade_process.rs
@@ -13,12 +13,19 @@ pub struct CloseUpgradeProcess {
 #[derive(Debug, PartialEq, Eq, Clone, Hash, serde::Serialize, serde::Deserialize)]
 pub struct CloseUpgradeProcessInstructionAccounts {
     pub funds_to: solana_pubkey::Pubkey,
-    pub starbase_and_starbase_player: solana_pubkey::Pubkey,
+    // StarbaseMutAndStarbasePlayer expansion
+    pub starbase: solana_pubkey::Pubkey,
+    pub starbase_player: solana_pubkey::Pubkey,
     pub resource_crafting_instance: solana_pubkey::Pubkey,
     pub resource_crafting_process: solana_pubkey::Pubkey,
     pub resource_recipe: solana_pubkey::Pubkey,
     pub resource_crafting_facility: solana_pubkey::Pubkey,
-    pub game_accounts_and_profile: solana_pubkey::Pubkey,
+    // GameAndGameStateAndProfile expansion
+    pub key: solana_pubkey::Pubkey,
+    pub profile: solana_pubkey::Pubkey,
+    pub profile_faction: solana_pubkey::Pubkey,
+    pub game_id: solana_pubkey::Pubkey,
+    pub game_state: solana_pubkey::Pubkey,
     pub crafting_program: solana_pubkey::Pubkey,
 }
 
@@ -30,22 +37,38 @@ impl carbon_core::deserialize::ArrangeAccounts for CloseUpgradeProcess {
     ) -> Option<Self::ArrangedAccounts> {
         let mut iter = accounts.iter();
         let funds_to = next_account(&mut iter)?;
-        let starbase_and_starbase_player = next_account(&mut iter)?;
+
+        // StarbaseMutAndStarbasePlayer expansion
+        let starbase = next_account(&mut iter)?;
+        let starbase_player = next_account(&mut iter)?;
+
         let resource_crafting_instance = next_account(&mut iter)?;
         let resource_crafting_process = next_account(&mut iter)?;
         let resource_recipe = next_account(&mut iter)?;
         let resource_crafting_facility = next_account(&mut iter)?;
-        let game_accounts_and_profile = next_account(&mut iter)?;
+
+        // GameAndGameStateAndProfile expansion
+        let key = next_account(&mut iter)?;
+        let profile = next_account(&mut iter)?;
+        let profile_faction = next_account(&mut iter)?;
+        let game_id = next_account(&mut iter)?;
+        let game_state = next_account(&mut iter)?;
+
         let crafting_program = next_account(&mut iter)?;
 
         Some(CloseUpgradeProcessInstructionAccounts {
             funds_to,
-            starbase_and_starbase_player,
+            starbase,
+            starbase_player,
             resource_crafting_instance,
             resource_crafting_process,
             resource_recipe,
             resource_crafting_facility,
-            game_accounts_and_profile,
+            key,
+            profile,
+            profile_faction,
+            game_id,
+            game_state,
             crafting_program,
         })
     }
diff --git a/src/instructions/complete_starbase_upgrade.rs b/src/instructions/complete_starbase_upgrade.rs
index 956b940..771562a 100644
--- a/src/instructions/complete_starbase_upgrade.rs
+++ b/src/instructions/complete_starbase_upgrade.rs
@@ -13,13 +13,20 @@ pub struct CompleteStarbaseUpgrade {
 #[derive(Debug, PartialEq, Eq, Clone, Hash, serde::Serialize, serde::Deserialize)]
 pub struct CompleteStarbaseUpgradeInstructionAccounts {
     pub funder: solana_pubkey::Pubkey,
-    pub starbase_and_starbase_player: solana_pubkey::Pubkey,
+    // StarbaseMutAndStarbasePlayer expansion
+    pub starbase: solana_pubkey::Pubkey,
+    pub starbase_player: solana_pubkey::Pubkey,
     pub crafting_facility: solana_pubkey::Pubkey,
     pub upgrade_facility: solana_pubkey::Pubkey,
     pub upgrade_recipe: solana_pubkey::Pubkey,
     pub new_recipe_category: solana_pubkey::Pubkey,
     pub crafting_domain: solana_pubkey::Pubkey,
-    pub game_accounts_and_profile: solana_pubkey::Pubkey,
+    // GameAndGameStateAndProfile expansion
+    pub key: solana_pubkey::Pubkey,
+    pub profile: solana_pubkey::Pubkey,
+    pub profile_faction: solana_pubkey::Pubkey,
+    pub game_id: solana_pubkey::Pubkey,
+    pub game_state: solana_pubkey::Pubkey,
     pub crafting_program: solana_pubkey::Pubkey,
     pub system_program: solana_pubkey::Pubkey,
 }
@@ -32,25 +39,41 @@ impl carbon_core::deserialize::ArrangeAccounts for CompleteStarbaseUpgrade {
     ) -> Option<Self::ArrangedAccounts> {
         let mut iter = accounts.iter();
         let funder = next_account(&mut iter)?;
-        let starbase_and_starbase_player = next_account(&mut iter)?;
+
+        // StarbaseMutAndStarbasePlayer expansion
+        let starbase = next_account(&mut iter)?;
+        let starbase_player = next_account(&mut iter)?;
+
         let crafting_facility = next_account(&mut iter)?;
         let upgrade_facility = next_account(&mut iter)?;
         let upgrade_recipe = next_account(&mut iter)?;
         let new_recipe_category = next_account(&mut iter)?;
         let crafting_domain = next_account(&mut iter)?;
-        let game_accounts_and_profile = next_account(&mut iter)?;
+
+        // GameAndGameStateAndProfile expansion
+        let key = next_account(&mut iter)?;
+        let profile = next_account(&mut iter)?;
+        let profile_faction = next_account(&mut iter)?;
+        let game_id = next_account(&mut iter)?;
+        let game_state = next_account(&mut iter)?;
+
         let crafting_program = next_account(&mut iter)?;
         let system_program = next_account(&mut iter)?;
 
         Some(CompleteStarbaseUpgradeInstructionAccounts {
             funder,
-            starbase_and_starbase_player,
+            starbase,
+            starbase_player,
             crafting_facility,
             upgrade_facility,
             upgrade_recipe,
             new_recipe_category,
             crafting_domain,
-            game_accounts_and_profile,
+            key,
+            profile,
+            profile_faction,
+            game_id,
+            game_state,
             crafting_program,
             system_program,
         })
diff --git a/src/instructions/create_cargo_pod.rs b/src/instructions/create_cargo_pod.rs
index a8d1961..8e2bb8b 100644
--- a/src/instructions/create_cargo_pod.rs
+++ b/src/instructions/create_cargo_pod.rs
@@ -13,10 +13,17 @@ pub struct CreateCargoPod {
 #[derive(Debug, PartialEq, Eq, Clone, Hash, serde::Serialize, serde::Deserialize)]
 pub struct CreateCargoPodInstructionAccounts {
     pub funder: solana_pubkey::Pubkey,
-    pub starbase_and_starbase_player: solana_pubkey::Pubkey,
+    // StarbaseMutAndStarbasePlayer expansion
+    pub starbase: solana_pubkey::Pubkey,
+    pub starbase_player: solana_pubkey::Pubkey,
     pub cargo_pod: solana_pubkey::Pubkey,
     pub cargo_stats_definition: solana_pubkey::Pubkey,
-    pub game_accounts_and_profile: solana_pubkey::Pubkey,
+    // GameAndGameStateAndProfile expansion
+    pub key: solana_pubkey::Pubkey,
+    pub profile: solana_pubkey::Pubkey,
+    pub profile_faction: solana_pubkey::Pubkey,
+    pub game_id: solana_pubkey::Pubkey,
+    pub game_state: solana_pubkey::Pubkey,
     pub cargo_program: solana_pubkey::Pubkey,
     pub system_program: solana_pubkey::Pubkey,
 }
@@ -29,19 +36,35 @@ impl carbon_core::deserialize::ArrangeAccounts for CreateCargoPod {
     ) -> Option<Self::ArrangedAccounts> {
         let mut iter = accounts.iter();
         let funder = next_account(&mut iter)?;
-        let starbase_and_starbase_player = next_account(&mut iter)?;
+
+        // StarbaseMutAndStarbasePlayer expansion
+        let starbase = next_account(&mut iter)?;
+        let starbase_player = next_account(&mut iter)?;
+
         let cargo_pod = next_account(&mut iter)?;
         let cargo_stats_definition = next_account(&mut iter)?;
-        let game_accounts_and_profile = next_account(&mut iter)?;
+
+        // GameAndGameStateAndProfile expansion
+        let key = next_account(&mut iter)?;
+        let profile = next_account(&mut iter)?;
+        let profile_faction = next_account(&mut iter)?;
+        let game_id = next_account(&mut iter)?;
+        let game_state = next_account(&mut iter)?;
+
         let cargo_program = next_account(&mut iter)?;
         let system_program = next_account(&mut iter)?;
 
         Some(CreateCargoPodInstructionAccounts {
             funder,
-            starbase_and_starbase_player,
+            starbase,
+            starbase_player,
             cargo_pod,
             cargo_stats_definition,
-            game_accounts_and_profile,
+            key,
+            profile,
+            profile_faction,
+            game_id,
+            game_state,
             cargo_program,
             system_program,
         })
diff --git a/src/instructions/create_starbase_upgrade_resource_process.rs b/src/instructions/create_starbase_upgrade_resource_process.rs
index e0f3cb7..570cb61 100644
--- a/src/instructions/create_starbase_upgrade_resource_process.rs
+++ b/src/instructions/create_starbase_upgrade_resource_process.rs
@@ -13,13 +13,20 @@ pub struct CreateStarbaseUpgradeResourceProcess {
 #[derive(Debug, PartialEq, Eq, Clone, Hash, serde::Serialize, serde::Deserialize)]
 pub struct CreateStarbaseUpgradeResourceProcessInstructionAccounts {
     pub funder: solana_pubkey::Pubkey,
-    pub starbase_and_starbase_player: solana_pubkey::Pubkey,
+    // StarbaseMutAndStarbasePlayer expansion
+    pub starbase: solana_pubkey::Pubkey,
+    pub starbase_player: solana_pubkey::Pubkey,
     pub crafting_instance: solana_pubkey::Pubkey,
     pub upgrade_facility: solana_pubkey::Pubkey,
     pub crafting_process: solana_pubkey::Pubkey,
     pub crafting_recipe: solana_pubkey::Pubkey,
     pub crafting_domain: solana_pubkey::Pubkey,
-    pub game_accounts_and_profile: solana_pubkey::Pubkey,
+    // GameAndGameStateAndProfile expansion
+    pub key: solana_pubkey::Pubkey,
+    pub profile: solana_pubkey::Pubkey,
+    pub profile_faction: solana_pubkey::Pubkey,
+    pub game_id: solana_pubkey::Pubkey,
+    pub game_state: solana_pubkey::Pubkey,
     pub crafting_program: solana_pubkey::Pubkey,
     pub system_program: solana_pubkey::Pubkey,
 }
@@ -32,25 +39,41 @@ impl carbon_core::deserialize::ArrangeAccounts for CreateStarbaseUpgradeResource
     ) -> Option<Self::ArrangedAccounts> {
         let mut iter = accounts.iter();
         let funder = next_account(&mut iter)?;
-        let starbase_and_starbase_player = next_account(&mut iter)?;
+
+        // StarbaseMutAndStarbasePlayer expansion
+        let starbase = next_account(&mut iter)?;
+        let starbase_player = next_account(&mut iter)?;
+
         let crafting_instance = next_account(&mut iter)?;
         let upgrade_facility = next_account(&mut iter)?;
         let crafting_process = next_account(&mut iter)?;
         let crafting_recipe = next_account(&mut iter)?;
         let crafting_domain = next_account(&mut iter)?;
-        let game_accounts_and_profile = next_account(&mut iter)?;
+
+        // GameAndGameStateAndProfile expansion
+        let key = next_account(&mut iter)?;
+        let profile = next_account(&mut iter)?;
+        let profile_faction = next_account(&mut iter)?;
+        let game_id = next_account(&mut iter)?;
+        let game_state = next_account(&mut iter)?;
+
         let crafting_program = next_account(&mut iter)?;
         let system_program = next_account(&mut iter)?;
 
         Some(CreateStarbaseUpgradeResourceProcessInstructionAccounts {
             funder,
-            starbase_and_starbase_player,
+            starbase,
+            starbase_player,
             crafting_instance,
             upgrade_facility,
             crafting_process,
             crafting_recipe,
             crafting_domain,
-            game_accounts_and_profile,
+            key,
+            profile,
+            profile_faction,
+            game_id,
+            game_state,
             crafting_program,
             system_program,
         })
diff --git a/src/instructions/deposit_starbase_upkeep_resource.rs b/src/instructions/deposit_starbase_upkeep_resource.rs
index 1a3fe5d..f3f4a0b 100644
--- a/src/instructions/deposit_starbase_upkeep_resource.rs
+++ b/src/instructions/deposit_starbase_upkeep_resource.rs
@@ -13,15 +13,25 @@ pub struct DepositStarbaseUpkeepResource {
 #[derive(Debug, PartialEq, Eq, Clone, Hash, serde::Serialize, serde::Deserialize)]
 pub struct DepositStarbaseUpkeepResourceInstructionAccounts {
     pub funds_to: solana_pubkey::Pubkey,
-    pub starbase_and_starbase_player: solana_pubkey::Pubkey,
+    // StarbaseMutAndStarbasePlayer expansion
+    pub starbase: solana_pubkey::Pubkey,
+    pub starbase_player: solana_pubkey::Pubkey,
     pub cargo_pod_from: solana_pubkey::Pubkey,
     pub cargo_type: solana_pubkey::Pubkey,
     pub cargo_stats_definition: solana_pubkey::Pubkey,
     pub token_from: solana_pubkey::Pubkey,
     pub token_mint: solana_pubkey::Pubkey,
-    pub game_accounts_and_profile: solana_pubkey::Pubkey,
+    // GameAndGameStateAndProfile expansion
+    pub key: solana_pubkey::Pubkey,
+    pub profile: solana_pubkey::Pubkey,
+    pub profile_faction: solana_pubkey::Pubkey,
+    pub game_id: solana_pubkey::Pubkey,
+    pub game_state: solana_pubkey::Pubkey,
     pub resource_recipe: solana_pubkey::Pubkey,
-    pub loyalty_points_accounts: solana_pubkey::Pubkey,
+    // PointsModificationAccounts expansion (loyalty)
+    pub loyalty_user_points_account: solana_pubkey::Pubkey,
+    pub loyalty_points_category: solana_pubkey::Pubkey,
+    pub loyalty_points_modifier_account: solana_pubkey::Pubkey,
     pub progression_config: solana_pubkey::Pubkey,
     pub points_program: solana_pubkey::Pubkey,
     pub cargo_program: solana_pubkey::Pubkey,
@@ -36,15 +46,31 @@ impl carbon_core::deserialize::ArrangeAccounts for DepositStarbaseUpkeepResource
     ) -> Option<Self::ArrangedAccounts> {
         let mut iter = accounts.iter();
         let funds_to = next_account(&mut iter)?;
-        let starbase_and_starbase_player = next_account(&mut iter)?;
+
+        // StarbaseMutAndStarbasePlayer expansion
+        let starbase = next_account(&mut iter)?;
+        let starbase_player = next_account(&mut iter)?;
+
         let cargo_pod_from = next_account(&mut iter)?;
         let cargo_type = next_account(&mut iter)?;
         let cargo_stats_definition = next_account(&mut iter)?;
         let token_from = next_account(&mut iter)?;
         let token_mint = next_account(&mut iter)?;
-        let game_accounts_and_profile = next_account(&mut iter)?;
+
+        // GameAndGameStateAndProfile expansion
+        let key = next_account(&mut iter)?;
+        let profile = next_account(&mut iter)?;
+        let profile_faction = next_account(&mut iter)?;
+        let game_id = next_account(&mut iter)?;
+        let game_state = next_account(&mut iter)?;
+
         let resource_recipe = next_account(&mut iter)?;
-        let loyalty_points_accounts = next_account(&mut iter)?;
+
+        // PointsModificationAccounts expansion (loyalty)
+        let loyalty_user_points_account = next_account(&mut iter)?;
+        let loyalty_points_category = next_account(&mut iter)?;
+        let loyalty_points_modifier_account = next_account(&mut iter)?;
+
         let progression_config = next_account(&mut iter)?;
         let points_program = next_account(&mut iter)?;
         let cargo_program = next_account(&mut iter)?;
@@ -52,15 +78,22 @@ impl carbon_core::deserialize::ArrangeAccounts for DepositStarbaseUpkeepResource
 
         Some(DepositStarbaseUpkeepResourceInstructionAccounts {
             funds_to,
-            starbase_and_starbase_player,
+            starbase,
+            starbase_player,
             cargo_pod_from,
             cargo_type,
             cargo_stats_definition,
             token_from,
             token_mint,
-            game_accounts_and_profile,
+            key,
+            profile,
+            profile_faction,
+            game_id,
+            game_state,
             resource_recipe,
-            loyalty_points_accounts,
+            loyalty_user_points_account,
+            loyalty_points_category,
+            loyalty_points_modifier_account,
             progression_config,
             points_program,
             cargo_program,
diff --git a/src/instructions/start_starbase_upgrade.rs b/src/instructions/start_starbase_upgrade.rs
index ca1a93a..5006403 100644
--- a/src/instructions/start_starbase_upgrade.rs
+++ b/src/instructions/start_starbase_upgrade.rs
@@ -13,10 +13,17 @@ pub struct StartStarbaseUpgrade {
 #[derive(Debug, PartialEq, Eq, Clone, Hash, serde::Serialize, serde::Deserialize)]
 pub struct StartStarbaseUpgradeInstructionAccounts {
     pub funder: solana_pubkey::Pubkey,
-    pub starbase_and_starbase_player: solana_pubkey::Pubkey,
+    // StarbaseMutAndStarbasePlayer expansion
+    pub starbase: solana_pubkey::Pubkey,
+    pub starbase_player: solana_pubkey::Pubkey,
     pub upgrade_facility: solana_pubkey::Pubkey,
     pub upgrade_recipe: solana_pubkey::Pubkey,
-    pub game_accounts_and_profile: solana_pubkey::Pubkey,
+    // GameAndGameStateAndProfile expansion
+    pub key: solana_pubkey::Pubkey,
+    pub profile: solana_pubkey::Pubkey,
+    pub profile_faction: solana_pubkey::Pubkey,
+    pub game_id: solana_pubkey::Pubkey,
+    pub game_state: solana_pubkey::Pubkey,
     pub system_program: solana_pubkey::Pubkey,
 }
 
@@ -28,18 +35,34 @@ impl carbon_core::deserialize::ArrangeAccounts for StartStarbaseUpgrade {
     ) -> Option<Self::ArrangedAccounts> {
         let mut iter = accounts.iter();
         let funder = next_account(&mut iter)?;
-        let starbase_and_starbase_player = next_account(&mut iter)?;
+
+        // StarbaseMutAndStarbasePlayer expansion
+        let starbase = next_account(&mut iter)?;
+        let starbase_player = next_account(&mut iter)?;
+
         let upgrade_facility = next_account(&mut iter)?;
         let upgrade_recipe = next_account(&mut iter)?;
-        let game_accounts_and_profile = next_account(&mut iter)?;
+
+        // GameAndGameStateAndProfile expansion
+        let key = next_account(&mut iter)?;
+        let profile = next_account(&mut iter)?;
+        let profile_faction = next_account(&mut iter)?;
+        let game_id = next_account(&mut iter)?;
+        let game_state = next_account(&mut iter)?;
+
         let system_program = next_account(&mut iter)?;
 
         Some(StartStarbaseUpgradeInstructionAccounts {
             funder,
-            starbase_and_starbase_player,
+            starbase,
+            starbase_player,
             upgrade_facility,
             upgrade_recipe,
-            game_accounts_and_profile,
+            key,
+            profile,
+            profile_faction,
+            game_id,
+            game_state,
             system_program,
         })
     }
diff --git a/src/instructions/submit_starbase_upgrade_resource.rs b/src/instructions/submit_starbase_upgrade_resource.rs
index d740f42..bb8dae5 100644
--- a/src/instructions/submit_starbase_upgrade_resource.rs
+++ b/src/instructions/submit_starbase_upgrade_resource.rs
@@ -13,7 +13,9 @@ pub struct SubmitStarbaseUpgradeResource {
 #[derive(Debug, PartialEq, Eq, Clone, Hash, serde::Serialize, serde::Deserialize)]
 pub struct SubmitStarbaseUpgradeResourceInstructionAccounts {
     pub funds_to: solana_pubkey::Pubkey,
-    pub starbase_and_starbase_player: solana_pubkey::Pubkey,
+    // StarbaseMutAndStarbasePlayer expansion
+    pub starbase: solana_pubkey::Pubkey,
+    pub starbase_player: solana_pubkey::Pubkey,
     pub resource_crafting_instance: solana_pubkey::Pubkey,
     pub resource_crafting_process: solana_pubkey::Pubkey,
     pub resource_crafting_facility: solana_pubkey::Pubkey,
@@ -26,8 +28,16 @@ pub struct SubmitStarbaseUpgradeResourceInstructionAccounts {
     pub token_from: solana_pubkey::Pubkey,
     pub token_to: solana_pubkey::Pubkey,
     pub token_mint: solana_pubkey::Pubkey,
-    pub game_accounts_and_profile: solana_pubkey::Pubkey,
-    pub loyalty_points_accounts: solana_pubkey::Pubkey,
+    // GameAndGameStateAndProfile expansion
+    pub key: solana_pubkey::Pubkey,
+    pub profile: solana_pubkey::Pubkey,
+    pub profile_faction: solana_pubkey::Pubkey,
+    pub game_id: solana_pubkey::Pubkey,
+    pub game_state: solana_pubkey::Pubkey,
+    // PointsModificationAccounts expansion (loyalty)
+    pub loyalty_user_points_account: solana_pubkey::Pubkey,
+    pub loyalty_points_category: solana_pubkey::Pubkey,
+    pub loyalty_points_modifier_account: solana_pubkey::Pubkey,
     pub progression_config: solana_pubkey::Pubkey,
     pub points_program: solana_pubkey::Pubkey,
     pub crafting_program: solana_pubkey::Pubkey,
@@ -43,7 +53,11 @@ impl carbon_core::deserialize::ArrangeAccounts for SubmitStarbaseUpgradeResource
     ) -> Option<Self::ArrangedAccounts> {
         let mut iter = accounts.iter();
         let funds_to = next_account(&mut iter)?;
-        let starbase_and_starbase_player = next_account(&mut iter)?;
+
+        // StarbaseMutAndStarbasePlayer expansion
+        let starbase = next_account(&mut iter)?;
+        let starbase_player = next_account(&mut iter)?;
+
         let resource_crafting_instance = next_account(&mut iter)?;
         let resource_crafting_process = next_account(&mut iter)?;
         let resource_crafting_facility = next_account(&mut iter)?;
@@ -56,8 +70,19 @@ impl carbon_core::deserialize::ArrangeAccounts for SubmitStarbaseUpgradeResource
         let token_from = next_account(&mut iter)?;
         let token_to = next_account(&mut iter)?;
         let token_mint = next_account(&mut iter)?;
-        let game_accounts_and_profile = next_account(&mut iter)?;
-        let loyalty_points_accounts = next_account(&mut iter)?;
+
+        // GameAndGameStateAndProfile expansion
+        let key = next_account(&mut iter)?;
+        let profile = next_account(&mut iter)?;
+        let profile_faction = next_account(&mut iter)?;
+        let game_id = next_account(&mut iter)?;
+        let game_state = next_account(&mut iter)?;
+
+        // PointsModificationAccounts expansion (loyalty)
+        let loyalty_user_points_account = next_account(&mut iter)?;
+        let loyalty_points_category = next_account(&mut iter)?;
+        let loyalty_points_modifier_account = next_account(&mut iter)?;
+
         let progression_config = next_account(&mut iter)?;
         let points_program = next_account(&mut iter)?;
         let crafting_program = next_account(&mut iter)?;
@@ -66,7 +91,8 @@ impl carbon_core::deserialize::ArrangeAccounts for SubmitStarbaseUpgradeResource
 
         Some(SubmitStarbaseUpgradeResourceInstructionAccounts {
             funds_to,
-            starbase_and_starbase_player,
+            starbase,
+            starbase_player,
             resource_crafting_instance,
             resource_crafting_process,
             resource_crafting_facility,
@@ -79,8 +105,14 @@ impl carbon_core::deserialize::ArrangeAccounts for SubmitStarbaseUpgradeResource
             token_from,
             token_to,
             token_mint,
-            game_accounts_and_profile,
-            loyalty_points_accounts,
+            key,
+            profile,
+            profile_faction,
+            game_id,
+            game_state,
+            loyalty_user_points_account,
+            loyalty_points_category,
+            loyalty_points_modifier_account,
             progression_config,
             points_program,
             crafting_program,
diff --git a/src/instructions/sync_starbase_upgrade_ingredients.rs b/src/instructions/sync_starbase_upgrade_ingredients.rs
index 281226c..8eb9704 100644
--- a/src/instructions/sync_starbase_upgrade_ingredients.rs
+++ b/src/instructions/sync_starbase_upgrade_ingredients.rs
@@ -15,7 +15,12 @@ pub struct SyncStarbaseUpgradeIngredientsInstructionAccounts {
     pub funder: solana_pubkey::Pubkey,
     pub starbase: solana_pubkey::Pubkey,
     pub upgrade_recipe: solana_pubkey::Pubkey,
-    pub game_accounts_and_profile: solana_pubkey::Pubkey,
+    // GameAndGameStateAndProfile expansion
+    pub key: solana_pubkey::Pubkey,
+    pub profile: solana_pubkey::Pubkey,
+    pub profile_faction: solana_pubkey::Pubkey,
+    pub game_id: solana_pubkey::Pubkey,
+    pub game_state: solana_pubkey::Pubkey,
     pub system_program: solana_pubkey::Pubkey,
 }
 
@@ -29,14 +34,25 @@ impl carbon_core::deserialize::ArrangeAccounts for SyncStarbaseUpgradeIngredient
         let funder = next_account(&mut iter)?;
         let starbase = next_account(&mut iter)?;
         let upgrade_recipe = next_account(&mut iter)?;
-        let game_accounts_and_profile = next_account(&mut iter)?;
+
+        // GameAndGameStateAndProfile expansion
+        let key = next_account(&mut iter)?;
+        let profile = next_account(&mut iter)?;
+        let profile_faction = next_account(&mut iter)?;
+        let game_id = next_account(&mut iter)?;
+        let game_state = next_account(&mut iter)?;
+
         let system_program = next_account(&mut iter)?;
 
         Some(SyncStarbaseUpgradeIngredientsInstructionAccounts {
             funder,
             starbase,
             upgrade_recipe,
-            game_accounts_and_profile,
+            key,
+            profile,
+            profile_faction,
+            game_id,
+            game_state,
             system_program,
         })
     }
