diff --git a/src/instructions/mod.rs b/src/instructions/mod.rs
index 82ef02f..72d0b5d 100644
--- a/src/instructions/mod.rs
+++ b/src/instructions/mod.rs
@@ -48,7 +48,7 @@ pub use self::tcomp_noop::*;
 pub use self::withdraw_fees::*;
 
 #[derive(Debug, Clone, PartialEq)]
-#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "serde", derive(carbon_core::InstructionType, serde::Serialize, serde::Deserialize))]
 #[cfg_attr(feature = "serde", serde(tag = "type", content = "data"))]
 pub enum TcompInstruction {
     Bid(Bid),
